nni/algorithms/compression/v2/pytorch/pruning/auto_compress_pruner.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from pathlib import Path
+from typing import Dict, List, Callable, Optional
+
+from torch import Tensor
+from torch.nn import Module
+
+from .basic_pruner import ADMMPruner
+from .iterative_pruner import IterativePruner, SimulatedAnnealingPruner
+from .tools import LotteryTicketTaskGenerator
+
+
+class AutoCompressTaskGenerator(LotteryTicketTaskGenerator):
+    def __init__(self, total_iteration: int, origin_model: Module, origin_config_list: List[Dict],
+                 origin_masks: Dict[str, Dict[str, Tensor]] = {}, sa_params: Dict = {}, log_dir: str = '.',
+                 keep_intermediate_result: bool = False):
+        self.iterative_pruner = SimulatedAnnealingPruner(model=None,
+                                                         config_list=None,
+                                                         log_dir=Path(log_dir, 'SA'),
+                                                         **sa_params)
+        super().__init__(total_iteration=total_iteration,
+                         origin_model=origin_model,
+                         origin_config_list=origin_config_list,
+                         origin_masks=origin_masks,
+                         log_dir=log_dir,
+                         keep_intermediate_result=keep_intermediate_result)
+
+    def _iterative_pruner_reset(self, model: Module, config_list: List[Dict] = [], masks: Dict[str, Dict[str, Tensor]] = {}):
+        self.iterative_pruner.task_generator._log_dir = Path(self._log_dir_root, 'SA')
+        self.iterative_pruner.reset(model, config_list=config_list, masks=masks)
+
+    def allocate_sparsity(self, new_config_list: List[Dict], model: Module, masks: Dict[str, Dict[str, Tensor]]):
+        self._iterative_pruner_reset(model, new_config_list, masks)
+        self.iterative_pruner.compress()
+        _, _, _, _, config_list = self.iterative_pruner.get_best_result()
+        return config_list
+
+
+class AutoCompressPruner(IterativePruner):
+    """
+    Parameters
+    ----------
+    model : Module
+        The origin unwrapped pytorch model to be pruned.
+    config_list : List[Dict]
+        The origin config list provided by the user. Note that this config_list is directly config the origin model.
+        This means the sparsity provided by the origin_masks should also be recorded in the origin_config_list.
+    total_iteration : int
+        The total iteration number.
+    evaluator : Callable[[Module], float]
+        Evaluate the pruned model and give a score.
+    admm_params : Dict
+        The parameters pass to the ADMMPruner.
+
+        - trainer : Callable[[Module, Optimizer, Callable].
```

Comment:
why not simply provide a link to ADMMPruner?

Reply:
because not all `ADMMPruner` parameters are expected to be set.